sql-plugin/src/main/scala/org/apache/spark/sql/rapids/execution/GpuHashJoin.scala

```diff
@@ -507,6 +510,141 @@ class ConditionalHashJoinIterator(
   }
 }
 
+
+class ExistenceJoinIterator(
+  spillableBuiltBatch: LazySpillableColumnarBatch,
+  boundBuildKeys: Seq[GpuExpression],
+  lazyStream: Iterator[LazySpillableColumnarBatch],
+  boundStreamKeys: Seq[GpuExpression],
+  boundCondition: Option[GpuExpression],
+  numFirstConditionTableColumns: Int,
+  compareNullsEqual: Boolean,
+  opTime: GpuMetric,
+  joinTime: GpuMetric
+) extends Iterator[ColumnarBatch]()
+    with TaskAutoCloseableResource with Arm {
+
+  use(spillableBuiltBatch)
+  val compiledConditionRes: Option[CompiledExpression] = boundCondition
+    .map(gpuExpr => use(gpuExpr.convertToAst(numFirstConditionTableColumns).compile()))
+
+  override def hasNext: Boolean = {
+    val streamHasNext = lazyStream.hasNext
+    if (!streamHasNext) {
+      close()
+    }
+    streamHasNext
+  }
+
+  override def next(): ColumnarBatch = {
+    withResource(lazyStream.next()) { lazyBatch =>
+      existenceJoinNextBatch(lazyBatch.getBatch)
+    }
```

Comment:
Just before returning a new output batch, we need to call `spillableBuiltBatch.allowSpilling()`, since we're holding onto that GPU resource while downstream nodes continue to process on the GPU.  `LazySpillableColumnarBatch` isn't actually spillable until that is called (and we don't want to call it before we actually need to support spilling).